### ISSUE TITLE
Ability to post audio (aka video files without a video track)

### ISSUE THREAD

**Issue by shodanx2** [LemmyNet/lemmy-ui#2529]
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

Add audio files in addition to the ability to post images and video

**Comment by dessalines** [comment]
Lemmy isn't a file sharing service, it stores no media of its own. Pictrs only supports pictures and video.

For this, you should use torrent magnet links, which lemmy supports both as post urls, or within comments.

**Comment by shodanx2** [comment]
Ok, well if it supports video, then repackaging the audio track in an mp4 file and adding a 320x240 all black pixels video stream will do ..... It is quite cumbersome but for the user it's click and play, versus a torrent link that would click and do nothing.

**Comment by shodanx2** [comment]
Files such as this one 


https://lemmy.ml/api/v3/image_proxy?url=https%3A%2F%2Ffiles.catbox.moe%2Fu3o241.mp4

<img width="1247" height="469" alt="Image" src="https://lemmy.ml/api/v3/image_proxy?url=https%3A%2F%2Ffiles.catbox.moe%2Fu3o241.mp4" />
https://lemmy.ml/post/36779575/21330283

Are welcomed on Lemmy, but audio files are not. ?

I don't understand the prejudice against sound that Big Tech has !

Bits are bits, it doesn't matter if they represent text, image, video or sound ! As long as they do not exceed the file limit it should be fine !

**Comment by Nutomic** [comment]
Lemmy is not big tech, its the exact opposite. Thing is that we are using [pict-rs](https://git.asonix.dog/asonix/pict-rs/) for image storage which does not support audio files (but you can ask the developer about it). Handling audio files directly in Lemmy without pict-rs would require too much work and we dont have the resources for it. That said you can always upload your audio file on an external site and put the link into your Lemmy post.

**Comment by shodanx2** [comment]
I understand Lemmy is not a product of Big Tech, however what I mean is that the general "good" practices of Big Tech and IT end up proliferating everywhere. And that translates into a widespread bias against audio only content in social media spaces unless they're explicitely about audio. 

Audio is basically forgotten about because nobody uses it and nobody uses it because it's forgotten and very inconvenient to use (compared with mp4,webm video files)

I understand that the filesharing backend can handle audio files, because this backend thinks it is an image distribution service.

I think it can be used to accept audio files by changing the MIME whitelist, but I'll have to dig more into it when I create my first Lemmy instance.

However, in the frontend, would there even be a html 5.0 audio media player if an mp3 file where posted ?

I checked github, it's also missing this functionality, even though the browser can do it without any special code

https://dn721609.ca.archive.org/0/items/llamawhippinintrobydjmikellama/demo.mp3

[demo.mp3](https://github.com/user-attachments/files/22595315/demo.mp3)

So, I think pict-rs can be cajolled into not rejecting audio files outright
However the lemmy web ui frontend should display the little default audio player when an audio file is linked to, instead of remaining a plain URL as shown above

**Comment by Nutomic** [comment]
> However the lemmy web ui frontend should display the little default audio player when an audio file is linked to, instead of remaining a plain URL as shown above

Youre right this is actually very easy: https://github.com/LemmyNet/lemmy-ui/pull/3500

**Comment by shodanx2** [comment]
cool !
